usersession/agent/session_agent_test.go

```diff
@@ -103,3 +110,72 @@ func (s *sessionAgentSuite) TestDying(c *C) {
 		c.Error("agent.Dying() channel was not closed when agent stopped")
 	}
 }
+
+func (s *sessionAgentSuite) TestExitOnIdle(c *C) {
+	agent, err := agent.New()
+	c.Assert(err, IsNil)
+	agent.IdleTimeout = 100 * time.Millisecond
+	startTime := time.Now()
+	agent.Start()
+	defer agent.Stop()
+
+	makeRequest := func() {
+		response, err := s.client.Get("http://localhost/v1/session-info")
+		c.Assert(err, IsNil)
+		defer response.Body.Close()
+		c.Check(response.StatusCode, Equals, 200)
+	}
+	makeRequest()
+	time.Sleep(25 * time.Millisecond)
+	makeRequest()
```

Comment:
how long does it take from Start to here, just wondering if 100ms is ok or low for a overloaded test machine

Reply:
The two requests plus the sleep generally take 27-27.5ms on my system, so about 1ms per request.  I haven't noticed any runs exceeding the idle timeout yet, but I suppose we could increase it.  That adds to the runtime of the test suite though.

Reply:
let's push it then to 150ms  (a factor of ~5 pessimism)